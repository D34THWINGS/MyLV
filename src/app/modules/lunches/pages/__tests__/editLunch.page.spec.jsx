import React from 'react'
import { shallow } from 'enzyme'
import { shallowToJson } from 'enzyme-to-json'

import { EditLunchPage } from '../editLunch.page'

jest.unmock('../editLunch.page')

describe('EditLunchPage', () => {
  let props

  const getWrapper = () => shallow(<EditLunchPage {...props} />)

  beforeEach(() => {
    props = {
      match: {
        params: {
          id: 'foo',
        },
      },
      lunch: {
        label: 'hello',
        date: '2012-01-01',
      },
      putLunch: jest.fn(),
      fetchLunchDetails: jest.fn(),
      isLoading: false,
    }
  })

  it('should render loading screen if loading', () => {
    // Given
    props.isLoading = true

    // When
    const wrapper = getWrapper()

    // Then
    expect(shallowToJson(wrapper)).toMatchSnapshot()
  })

  it('should render edit form if not loading', () => {
    // When
    const wrapper = getWrapper()

    // Then
    expect(shallowToJson(wrapper)).toMatchSnapshot()
  })

  it('should pass render function to edit form', () => {
    // When
    const wrapper = shallow(getWrapper().prop('render')({ children: <div>Foo</div> }))

    // Then
    expect(shallowToJson(wrapper)).toMatchSnapshot()
  })
})
