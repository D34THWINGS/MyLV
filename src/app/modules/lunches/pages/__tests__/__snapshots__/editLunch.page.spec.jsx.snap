// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EditLunchPage should pass render function to edit form 1`] = `
<WithStyles(Paper)
  elevation={2}
>
  <WithStyles(CardContent)>
    <WithStyles(Typography)
      component="h2"
      gutterBottom={true}
      variant="headline"
    >
      Modification de : 
      hello
    </WithStyles(Typography)>
    <div>
      Foo
    </div>
  </WithStyles(CardContent)>
  <WithStyles(CardActions)>
    <WithStyles(Button)
      color="primary"
      disabled={true}
      size="small"
      type="submit"
    >
      Enregistrer
    </WithStyles(Button)>
  </WithStyles(CardActions)>
</WithStyles(Paper)>
`;

exports[`EditLunchPage should render edit form if not loading 1`] = `
<WithStyles(ReduxForm)
  initialValues={
    Object {
      "date": "2012-01-01",
      "label": "hello",
    }
  }
  onFormSubmit={[MockFunction]}
  render={[Function]}
/>
`;

exports[`EditLunchPage should render loading screen if loading 1`] = `<WithStyles(LoadingPage) />`;
